### ISSUE TITLE
[False Positive] `index out of bound` warning when using `slice::to_vec()`

### ISSUE THREAD

**Issue by rikki322** [lizhuohua/rust-mir-checker#23]
`MirChecker` reports a false positive "index out of bound" warning when a vector is initialized using `slice::to_vec()`. However, the equivalent logic using the `vec!` macro works correctly without warnings.
**Reproduction**
```
#[allow(unused_variables)]
fn main() {
    // Case 1: Works as expected (No Warning)
    // MirChecker correctly tracks the length here.
    // let a = vec![1, 2, 3, 4, 5]; 
    
    // Case 2: False Positive
    // MirChecker fails to track the length, treating it as unknown/zero.
    let a = [1, 2, 3, 4, 5].to_vec();
    
    let b = a[0]; // <--- Triggers warning
}
```
**Actual Behavior**
The tool reports an index out of bound error for valid access `a[0]`:
```
warning: [MirChecker] Possible error: index out of bound
  --> src/main.rs:11:13
   |
11 |     let b = a[0]; // <--- Triggers warning
   | 
```
**Expected Behavior** 
No warning should be reported. The source array `[1, 2, 3, 4, 5]` has a static length of 5, and `to_vec()` preserves this length. Accessing index `0` is safe.
